packages/venia-ui/lib/components/App/app.js

```diff
@@ -83,64 +68,45 @@ const App = props => {
             if (!toasts.get(errorToastId)) {
                 addToast(errorToastProps);
             }
-        }
-    }, [errors, handleDismiss]); // eslint-disable-line
-
-    const { app, closeDrawer } = props;
-    const { drawer, hasBeenOffline, isOnline, overlay } = app;
-    const cartIsOpen = drawer === 'cart';
-
-    useEffect(() => {
-        if (hasBeenOffline) {
-            if (isOnline) {
-                addToast({
-                    type: 'info',
-                    icon: OnlineIcon,
-                    message: 'You are online.',
-                    timeout: 3000
-                });
-            } else {
-                addToast({
-                    type: 'error',
-                    icon: OfflineIcon,
-                    message:
-                        'You are offline. Some features may be unavailable.',
-                    timeout: 3000
-                });
-            }
-        }
-    }, [addToast, hasBeenOffline, isOnline]);
+        },
+        [addToast, toasts]
+    );
+
+    const talonProps = useApp({
+        handleError,
+        handleIsOffline,
+        handleIsOnline,
+        markErrorHandled,
+        renderError,
+        unhandledErrors
+    });
```

Comment:
Most of these, aside from the online/offline handlers could probably be moved to `useErrorBoundary` or something other than `app`. That said, I chose not to do that in this PR to keep the changeset minimal.